sqlglot/dialects/tsql.py

```diff
@@ -358,13 +370,75 @@ class Parser(parser.Parser):
         STATEMENT_PARSERS = {
             **parser.Parser.STATEMENT_PARSERS,
             TokenType.END: lambda self: self._parse_command(),
+            TokenType.ALIAS: lambda self: self._parse_alias(),
         }
 
         LOG_BASE_FIRST = False
         LOG_DEFAULTS_TO_LN = True
 
         CONCAT_NULL_OUTPUTS_STRING = True
 
+        def _parse_commit_or_rollback(self) -> exp.Commit | exp.Rollback:
+            """Applies to SQL Server and Azure SQL Database
+                COMMIT [ { TRAN | TRANSACTION }
+                    [ transaction_name | @tran_name_variable ] ]
+                    [ WITH ( DELAYED_DURABILITY = { OFF | ON } ) ]
+                    [ ; ]
+
+                ROLLBACK { TRAN | TRANSACTION }
+                    [ transaction_name | @tran_name_variable
+                    | savepoint_name | @savepoint_variable ]
+                    [ ; ]
+
+            Returns:
+                exp.Commit | exp.Rollback: _description_
+            """
+            rollback = False
+            if self._prev.token_type == TokenType.ROLLBACK:
+                rollback = True
+
+            transaction = None
+            if self._match_texts({"TRAN", "TRANSACTION"}):
+                transaction = self._prev.text
+            txn_name = self._parse_id_var()
+
+            durability = None
+            if self._match_text_seq("WITH", "(", "DELAYED_DURABILITY", "="):
```

Comment:
don't do this, use real matching

self.match(TokenType.With):
self .match_l_paren() etc

Reply:
I could use a little help on this one, it falls apart on me as soon as I add self.match_l_paren().

Reply:
You could also write this as:

```python
if self._match_pair(TokenType.WITH, TokenType.L_PAREN):
    self._match_text_seq("DELAYED_DURABILITY")
    self._match(TokenType.EQ)
    ...
```

Reply:
That works.